pkg/engine/runtime/kubernetes_runtime.go

```diff
@@ -116,20 +117,21 @@ func (k *KubernetesRuntime) Apply(ctx context.Context, request *ApplyRequest) *A
 			}
 			if patchedObj, err := resource.Patch(ctx, planObj.GetName(), types.MergePatchType, patchBody, patchOptions); err == nil {
 				res = patchedObj
-				tryClientSide = false
 			} else {
-				log.Errorf("ServerSideDryRun Patch failed, err: %v", err)
-			}
-		}
-		if tryClientSide {
-			// Fall back to ClientSideDryRun
-			mergedPatch, err := jsonpatch.MergePatch([]byte(current), patchBody)
-			if err != nil {
-				return &ApplyResponse{nil, status.NewErrorStatus(err)}
-			}
-			res = &unstructured.Unstructured{}
-			if err = res.UnmarshalJSON(mergedPatch); err != nil {
-				return &ApplyResponse{nil, status.NewErrorStatus(err)}
+				// Fall back to ClientSideDryRun
+				log.Errorf("ServerSideDryRun patch %s failed, fall back to ClientSideDryRun; err: %v", planState.ID, err)
```

Comment:
This err msg will be printed twice
Duplicated Code in  L108

Reply:
I am afraid not, there are two conditions, one is live state is nil, it a creating verb; another live state is not nil, its a patch verb.
they are totally different entering logic.

> This err msg will be printed twice Duplicated Code in L108